geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexForPartitionedRegion.java

```diff
@@ -78,17 +87,9 @@ protected RepositoryManager createRepositoryManager() {
     // create PR fileAndChunkRegion, but not to create its buckets for now
     final String fileRegionName = createFileRegionName();
     PartitionAttributes partitionAttributes = dataRegion.getPartitionAttributes();
-
-
-    // create PR chunkRegion, but not to create its buckets for now
-
-    // we will create RegionDirectories on the fly when data comes in
-    HeterogeneousLuceneSerializer mapper = new HeterogeneousLuceneSerializer(getFieldNames());
-    PartitionedRepositoryManager partitionedRepositoryManager =
-        new PartitionedRepositoryManager(this, mapper);
     DM dm = this.cache.getInternalDistributedSystem().getDistributionManager();
     LuceneBucketListener lucenePrimaryBucketListener =
-        new LuceneBucketListener(partitionedRepositoryManager, dm);
+        new LuceneBucketListener((PartitionedRepositoryManager) partitionedRepositoryManager, dm);
```

Comment:
Having to cast is a bit of a bad smell here. It seems like either LuceneBucketListener should take an AbstractPartitionedRepositoryManager, or createLuceneListenersAndFileChunkRegions should have been passed in a PartitionRespositoryManager.